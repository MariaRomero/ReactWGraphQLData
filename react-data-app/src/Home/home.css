.container {
  height: calc(100vh - 0px);
}

.poke-list {
  overflow: auto;
  width: 40%;
  display: inline-block;
  height: calc(100% - 200px);
}

.poke-selected {
  height: 20px;
  padding: 6px 6px;
  display: block;
}
.avatar {
  height: 20px;
  padding: 0 6px;
}

.text {
  display: inline;
}
