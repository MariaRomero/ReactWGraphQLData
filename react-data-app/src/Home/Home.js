import React, { Fragment } from "react";
import ReactList from "react-list";
import { BrowserRouter as Router, Link } from "react-router-dom";
import "./home.css";

export const Home = props => {
  const { data } = props;

  return (
    <Fragment>
      <h1>Pokemon List</h1>
      <div className="container">
        <div className="poke-list">
          <ReactList
            itemRenderer={(index, key) => {
              return (
                // <Router>
                <Link
                  to={`/${data.pokemons[index]["id"]}`}
                  Home
                  className="poke-selected"
                  key={key}
                >
                  <img
                    className="avatar"
                    alt={data.pokemons[index]["name"]}
                    src={data.pokemons[index]["image"]}
                  />
                  <p className="text">
                    {data.pokemons[index]["name"]}-
                    {data.pokemons[index]["number"]}
                  </p>
                </Link>
                // </Router>
              );
            }}
            length={data.pokemons.length}
            type="simple"
          />
        </div>
      </div>
    </Fragment>
  );
};
