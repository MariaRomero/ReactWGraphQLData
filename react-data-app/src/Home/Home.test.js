import React from "react";
import { Home } from "./Home";
import { render, fireEvent, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
const wait = require("waait");

afterEach(cleanup);

it("Should display list of pokemons", async () => {
  const mock = {
    pokemons: [
      {
        id: "1",
        image: "https://img.pokemondb.net/artwork/bulbasaur.jpg",
        name: "Bulbasaur",
        number: "001",
        resistant: ["playing", "splashing"],
        types: ["fabric", "plastic"],
        weaknesses: ["Fire", "under 3", "water"]
      },
      {
        id: "2",
        image: "https://img.pokemondb.net/artwork/venusaur.jpg",
        name: "Venusaur",
        number: "003",
        resistant: ["Water", "Electric", "Grass", "Fighting", "Fairy"],
        types: ["Grass", "Poison"],
        weaknesses: ["Fire", "Ice", "Flying", "Psychic"]
      }
    ]
  };
  const { getByText, container } = render(<Home data={mock} />);
  await wait(0);
  expect(container).toMatchSnapshot();
  expect(getByText("Pokemon List")).toBeInTheDocument();
  expect(getByText("Bulbasaur-001")).toBeInTheDocument();
  expect(getByText("Venusaur-003")).toBeInTheDocument();
});
