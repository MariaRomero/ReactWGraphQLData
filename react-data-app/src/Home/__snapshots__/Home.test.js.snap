// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should display list of pokemons 1`] = `
<div>
  <h1>
    Pokemon List
  </h1>
  <div
    class="container"
  >
    <div
      class="poke-list"
    >
      <div>
        <a
          class="poke-selected"
          href="/1"
        >
          <img
            alt="Bulbasaur"
            class="avatar"
            src="https://img.pokemondb.net/artwork/bulbasaur.jpg"
          />
          <p
            class="text"
          >
            Bulbasaur
            -
            001
          </p>
        </a>
        <a
          class="poke-selected"
          href="/2"
        >
          <img
            alt="Venusaur"
            class="avatar"
            src="https://img.pokemondb.net/artwork/venusaur.jpg"
          />
          <p
            class="text"
          >
            Venusaur
            -
            003
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
